app/src/main/java/sdumchykov/androidApp/presentation/viewPager/myProfile/MyProfileFragment.kt

```diff
@@ -100,29 +110,29 @@ class MyProfileFragment :
 
     private fun imageViewMainProfilePictureSetOnClickListener() {
         binding.imageViewMainProfilePicture.setOnClickListener {
-//            activity?.viewModelStore.clear()
-
             val fetchContactList = !viewModel.getFetchContactList()
             viewModel.setFetchContactList(fetchContactList)
 
-            val toastText = if (fetchContactList) SHOW_CONTACT_LIST
+            if (fetchContactList) {
+                if (!isPermissionsGranted()) {
+                    FetchContacts().fetchContacts(
+                        activity as AppCompatActivity,
+                        { parentViewModel.initRealUsersList() },
+                        { parentViewModel.initHardcodedDataList() }
```

Comment:
Why do you initialize both of lists?

Reply:
I ran into the problem that under plain circumstances the code runs faster than the permission is granted. Therefore, I pass the function itself to the permission handler so that the flow of execution of instructions is more or less controlled.

Reply:
In short, I provide stable execution, although I understand that I sacrifice the quality of the code.